### ISSUE TITLE
Stop sending block templates to pool

### ISSUE THREAD

**Issue by luke-jr** [OCEAN-xyz/datum_gateway#1]
The current version of DATUM shares the templates with the pool, but it would be more robust if we didn't. For validity checks, we can share encrypted templates with other miners (keys verified by their PoW).

**Comment by chrisguida** [comment]
What is "share encrypted templates with other miners"? Can you expand on what this means? Are you saying that miners would validate each others' shares rather than Ocean validating them in a centralized way?

**Comment by luke-jr** [comment]
Exactly. Each miner chooses a key, and we commit to that key in their shares (so the pool can't trivially MITM), then pair up miners with each other so we can't see the templates, but still exchange them for cross validity checks.
